csrc/dynamic_transform.cpp

```diff
@@ -794,6 +795,19 @@ bool DynamicTransformConcretizer::propagateFromProducerToConsumer(
 
   auto def = consumer->definition();
 
+  std::vector<std::unordered_map<IterDomain*, IterDomain*>> c2p_maps;
```

Comment:
Why is this taken out of the loop below?

Reply:
Previously we were recomputing `PairwiseRootDomainMap root_map(producer, consumer)` for each root axis. This just precomputes each producer/consumer map once before iterating over the root domain.